src/activities/extract-meca.ts

```diff
@@ -73,21 +58,42 @@ export const extractMeca = async (version: VersionedReviewedPreprint): Promise<M
     Bucket: source.Bucket,
     Key: source.Key,
   };
-  const buffer = await s3.send(new GetObjectCommand(getObjectCommandInput))
-    .then((obj) => obj.Body?.transformToByteArray());
+  const data = await s3.send(new GetObjectCommand(getObjectCommandInput));
+
+  const downloadMeca = (body: Readable, localPath: string) => new Promise((resolve, reject) => {
```

Comment:
Notpick: This isn't very specific to a Meca file - should we move to a util (with a new name - maybe `downloadS3File` or something?)

Reply:
I will address this refactor this morning.

Thanks for checking it over.

Reply:
I have moved this to S3Bucket and renamed to `streamToFile`. I have tried to add a test but it's creating a lot of noise unrelated to adding a test just for this function. We should prioritise tests for this and other functions at another time.